server/sonar-web/src/main/js/api/users.ts

```diff
@@ -52,15 +63,40 @@ export function getUserGroups(login: string, organization?: string): Promise<any
 }
 
 export function getIdentityProviders(): Promise<{ identityProviders: IdentityProvider[] }> {
-  return getJSON('/api/users/identity_providers');
+  return getJSON('/api/users/identity_providers').catch(throwGlobalError);
 }
 
-export function searchUsers(query: string, pageSize?: number): Promise<any> {
-  const data: RequestData = { q: query };
-  if (pageSize != null) {
-    data.ps = pageSize;
-  }
-  return getJSON('/api/users/search', data);
+export function searchUsers(data: {
+  p?: number;
+  ps?: number;
+  q?: string;
+}): Promise<{ paging: Paging; users: User[] }> {
+  data.q = data.q || undefined;
+  return getJSON('/api/users/search', data).catch(throwGlobalError);
+}
+
+export function createUser(data: {
+  email?: string;
+  local?: boolean;
+  login: string;
+  name: string;
+  password?: string;
+  scmAccount: string[];
+}): Promise<void | Response> {
+  return post('/api/users/create', data);
+}
+
+export function updateUser(data: {
+  email?: string;
+  login: string;
+  name?: string;
+  scmAccount: string[];
+}): Promise<User> {
+  return postJSON('/api/users/update', data);
```

Comment:
I'm not adding the `throwGlobalError` there because the errors are displayed in the forms depending on if it's a 400 or something else.
See this : https://github.com/SonarSource/sonarqube/pull/2798/files/7a844d5ea5895c304c49c388f1db899c64628984..d36a31c66ef52706f1b0863e951ba0d07b96419b#diff-39b80814e35f38c23d0df2d3bfb54834R81